app/onboarding/page.tsx

```diff
@@ -0,0 +1,32 @@
+import { auth } from '@/auth';
+import { getUserById } from '@/lib/actions/user-actions';
+import OnboardingContainer from '@/components/onboarding/OnboardingContainer';
+import { Suspense } from 'react';
+import type { IUser } from '@/models/User';
+
+// ----------------------------------------------------------------
+
+const Onboarding = async () => {
+  const session = await auth();
+  if (!session?.user.id) return null;
+
+  let user: IUser | null = null;
+
+  try {
+    user = await getUserById(session.user.id);
+  } catch (error) {
+    console.log('Error fetching ', error);
+  }
+
+  if (!user) {
+    return null;
+  }
```

Comment:
```suggestion
  const user: IUser | null = await getUserById(session.user.id);
  
  if (!user) {
    return null;
  }
```

We can just fetch the user. If we get an error we don't have to catch it, instead we can create an `error` file which NextJS will use to show an error to the user. This is called an error boundary.

Reply:
Error file is like a page.tsx like layout.tsx just error.tsx ? thought to tackle this later when i polish the app.